### DISCUSSION TITLE
Role and responsibilities

### DISCUSSION THREAD

**Discussion by Harvester62** [camenduru/sd-webui-extension-records#1] [General]
Hi @ClashSAN I received your invitation to collaborate to this (what is this project exactly?) and I accepted it, but I am not sure about what my role and responsibility should be, what do you expect from me to do, etcetera.
I would appreciate it very much to receive some information, instructions and to understand if I can be effectively helpful with your expectations about my collaboration. I am open and available to listen and discuss my role with you.

Regards,
Riccardo Giovanetti
(Harvester62)

**Comment by Zyin055** [comment]
I would also like an official explanation from @ClashSAN

It seems like we are to update the README.md to include a working pair of versions for our custom extension and the A1111 Web UI.

**Comment by ClashSAN** [comment]
The aim for the "records" section is to provide at least 1 commit pairing where the extension works.

Feel free to add more, by date or by feature addition. You can put dropdowns inside dropdowns.

The rolling releases are the gameplan now. I don't want to encourage users to hop around different versions, and then the bugs don't get worked out. This is only an emergency option.

**Comment by Gerschel** [comment]
I get it. Sometimes it's a pita to keep having to maintain an extension. Sometimes you push an update to your extension to work with current release of the webui, but others might not be on current release, so it breaks for them. And the inverse, you might have stayed on a particular release of the webui, so you can focus on extending the extension instead of updating it daily, so you can make progress, with plans to adjust once you have the main pieces put together.  

A user can then say, they are not looking for latest, but most functional for their goal. I believe the overarching goal is for a user to say they will checkout a version, because they know these extensions (the ones they are interested in) have full support on them.  

Putting this together may be a bit of work, a table periodically, that shows which extensions work for that week. Which is where the extension writers come in. They submit if it works on version (for the week?).  

There may be a way to create an action where the repo owner can answer the question when they are pushing into their repo, and on a particular day each week, another action builds the table and commits it.

**Comment by Harvester62** [comment]
Thank you @ClashSAN for the clarification. I have (want) to keep my SDWUI updated to the latest commit because this seems the way Automatic1111 manages the project, and any Script and Extenstions should follow that strategy. 

A different project management approach might be thought, like making periodic releases, versions of SDWUI, instead of a continuous flow of changes to the main code which can quite frequently break Extensions to work properly.

I am not a developer nor a program manager but I am accustomed with the Blender release system: https://wiki.blender.org/wiki/Process/Release_Cycle#Release_Cycle (only for reference, inspiration)

For sure, such an approach to software release would give both the SDWUI and Extenstions developers the time to do things with some methodology, without hurrying and reducing the possibility to break things for the user, but this is only my personal opinion.
